File: go/problem32.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	fmt.Println(PandigitProducts())
}

func PandigitProducts() int {

	prodMap := make(map[int]int)
	prod := 1
	sum := 0

	for i := 2; i <= 100; i++ {

		for j := 100; j < 10000; j++ {

			prod = i * j
			pandigitNum := strconv.Itoa(i) + strconv.Itoa(j) + strconv.Itoa(i*j)

			if len(pandigitNum) != 9 {
				continue
			}

			if checkPandigitNumbers(pandigitNum) {
				if _, ok := prodMap[prod]; ok {
					continue
				} else {
					prodMap[prod] = 1
					sum += prod
				}

			}
		}

	}

	return sum
}

func checkPandigitNumbers(number string) bool {

	nums := []rune(number)
	var i rune
	for i = 49; i < 58; i++ { // 49 is 1
		if !contains(nums, i) {
			return false
		}
	}
	return true
  
}

func contains(s []rune, e rune) bool {

	for _, a := range s {
		if a == e {
			return true
		}
	}
	return false
  
}

File: python/problem32.py
def pandigitNumbers():
    prodSet = set()
    for i in range(2, 101):
        for j in range(100, 10001):
            prod = i * j
            pandigitNum = str(i) + str(j) + str(prod)
            if (len(pandigitNum) != 9): continue
            if checkPandigitNumber(pandigitNum):
                prodSet.add(prod)

    sum = 0
    for number in prodSet:
        sum += number

    return sum


def checkPandigitNumber(numAsStr):
    nums = list(numAsStr)
    for i in range(1, 10):
        if not ( str(i) in nums):
            return False
    return True


if __name__ == "__main__":
    print(pandigitNumbers())
